compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1394,15 +1394,30 @@ object Types {
      */
     def simplified(implicit ctx: Context) = ctx.simplify(this, null)
 
+    final def equals(that: Any, bs: BinderPairs): Boolean =
+      (this `eq` that.asInstanceOf[AnyRef]) || this.iso(that, bs)
```

Comment:
I'm not completely convinced that the default `equals` should call `iso`. Why not explicitly use `iso` in the few cases where we need structural equality (monitored subtyping checks, constraint handing, anything else ?)

Reply:
It's just a shorthand to avoid the repeated `eq` tests everywhere it is called.